ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -198,6 +225,39 @@ public void destroyAll(String owner) {
     }
   }
 
+  /**
+   * {@inheritDoc}
+   * Abnormally stopped service may lead to data corruption.
+   * Can take appropriate action by identifying state of service stopped.
+   */
+  @Override
+  public final synchronized boolean isClean() {
+    if (started) {
+      return clean;
+    }
+    if (rootDirectory.exists()) {
```

Comment:
Nobody should be calling anything on a stopped service (other than start). This should be able to be:
```java
if (started) {
  return clean;
} else {
  throw new IllegalStateException(...);
}
```